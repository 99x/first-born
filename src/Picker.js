import React, { Component } from "react";
import { StyleSheet, Platform, Picker, View } from "react-native";
import createReactClass from "create-react-class";
import Ionicon from "react-native-vector-icons/Ionicons";

class PickerAE extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false
        }
    }

    render() {
        const { color, ...otherProps } = this.props;

        const underLineColor = !color ? "#000" : color;

        if (Platform.OS === "android") {
            return (
                <View
                    style={this.state.focused ? [styles.pickerAndroid, { borderBottomColor: underLineColor, borderBottomWidth: 2 }] : [styles.pickerAndroid, { borderBottomColor: underLineColor, borderBottomWidth: 1 }]}
                    onResponderGrant={() => this.setState({ focused: true })}
                    onResponderTerminate={() => this.setState({ focused: false })}
                >
                    <Picker
                        ref={c => (this._root = c)}
                        mode="dropdown"
                        {...otherProps}>
                        {otherProps.children}
                    </Picker>
                    <Ionicon style={styles.icon} size={24} color={underLineColor} name={this.state.focused ? "md-arrow-dropup" : "md-arrow-dropdown"} />
                </View>
            )
        }
        return (
            <View style={styles.pickerIos}>
                <Picker
                    ref={c => (this._root = c)}
                    {...otherProps}>
                    {otherProps.children}
                </Picker>
            </View>
        )
    }
}

PickerAE.Item = createReactClass({
    render() {
        return <Picker.Item {...this.props} />;
    }
});

PickerAE.propTypes = {
    ...Picker.propTypes
}

const styles = StyleSheet.create({
    pickerAndroid: {
        width: "100%",
        borderTopRightRadius: 3,
        borderTopLeftRadius: 3,
        marginVertical: 10,
    },
    pickerIos: {
        backgroundColor: "white",
        borderRadius: 10,
        paddingHorizontal: 5,
        marginVertical: 10,
        width: '100%',
        borderColor: "#e2e2e2",
        borderWidth: 0.9,
        height: 50
    },
    icon: {
        position: "absolute",
        top: 13,
        right: 18
    }
});

export { PickerAE as Picker };
