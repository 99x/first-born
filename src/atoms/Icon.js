import { Platform } from "react-native";
import React, { Component } from "react";
import Ionicon from "react-native-vector-icons/Ionicons";
import PropTypes from "prop-types";
import { commonColors } from "../utils/color";

export class Icon extends Component {
    render() {
        const { name, size, color, ...otherProps } = this.props;

        if (!name.includes("logo")) {
            return (
                <Ionicon
                    name={
                        Platform.OS === "android" ? "md-" + name : "ios-" + name
                    }
                    size={size}
                    color={color}
                    {...otherProps}
                />
            );
        }
        return (
            <Ionicon name={name} size={size} color={color} {...otherProps} />
        );
    }
}

Icon.propTypes = {
    ...Ionicon.propTypes,
    name: PropTypes.string
};

Icon.defaultProps = {
    color: commonColors.white,
    size: 18
};
