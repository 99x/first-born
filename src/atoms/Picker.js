import React, { Component } from "react";
import { StyleSheet, Platform, Picker, View, Modal } from "react-native";
import createReactClass from "create-react-class";
import Ionicon from "react-native-vector-icons/Ionicons";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

class PickerAE extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            chosenValue: !props.placeholder && props.selectedValue ? props.selectedValue : undefined
        }
    }

    setValue = (value) => {
        this.setState({ chosenValue: value });
        if (this.props.onValueChange) {
            this.props.onValueChange(value);
        }
    }

    render() {
        const { color, placeholder, animationType, modalTransparent, selectedValue, children, ...otherProps } = this.props;
        const { focused, chosenValue } = this.state;

        if (Platform.OS === "android") {
            return (
                <View style={focused ? [styles.pickerAndroid, { borderColor: color, borderWidth: 2 }] : styles.pickerAndroid} >
                    <Picker
                        style={{ height: 45 }}
                        ref={c => (this._root = c)}
                        {...otherProps}
                        selectedValue={chosenValue}
                        onValueChange={this.setValue.bind(this)}
                    >
                        {children}
                    </Picker>
                </View>
            )
        }
        return (
            <View>
                <View style={focused ? [styles.pickerIos, { borderColor: color, borderWidth: 2 }] : styles.pickerIos}>
                    <Text
                        ref={(c) => this._root = c}
                        style={styles.input}
                        onPress={() => this.setState({ focused: true })}
                    >
                        {chosenValue ? chosenValue : placeholder}
                    </Text>
                    <Ionicon onPress={() => this.setState({ focused: true })} name={focused ? "ios-arrow-up" : "ios-arrow-down"} style={styles.icon} color={commonColors.inputGrey} size={24} />
                </View>
                <Modal
                    supportedOrientations={['portrait', 'landscape']}
                    animationType={animationType}
                    transparent={modalTransparent}
                    visible={focused}
                    onRequestClose={() => { }}
                >
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                    <Picker
                        ref={c => (this._root = c)}
                        {...otherProps}
                        selectedValue={chosenValue}
                        onValueChange={this.setValue.bind(this)}
                        style={{ flex: 1, backgroundColor: commonColors.white }}
                    >
                        {children}
                    </Picker>
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                </Modal>
            </View>
        )
    }
}

PickerAE.Item = createReactClass({
    render() {
        return <Picker.Item {...this.props} />;
    }
});

PickerAE.propTypes = {
    ...Picker.propTypes
}

PickerAE.defaultProps = {
    modalTransparent: true,
    color: commonColors.primary,
    animationType: "fade",
    mode: "dropdown",
    placeholder: "Select Option"
}

const styles = StyleSheet.create({
    pickerAndroid: {
        width: "100%",
        marginVertical: 10,
        borderRadius: 10,
        paddingHorizontal: 5,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9
    },
    pickerIos: {
        width: "100%",
        flexDirection: 'row',
        borderRadius: 15,
        marginVertical: 10,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9,
        justifyContent: 'center',
        alignItems: 'center',
        height: 45,
        paddingHorizontal: 5
    },
    icon: {
        position: "absolute",
        top: 13,
        right: 18
    },
    input: {
        flex: 1,
        paddingLeft: 10
    }
});

export { PickerAE as Picker };
