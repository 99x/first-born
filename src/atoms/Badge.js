import React, { Component } from "react";
import { View, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Badge extends Component {
    render() {
        const {
            color,
            children,
            ...otherProps
        } = this.props;

        const badgeColor = color;

        let badgeStyle = [styles.defaultBadge, {
            borderRadius: 15,
            backgroundColor: badgeColor
        }];

        let textStyle = {
            color: commonColors.white
        };

        let iconColor = commonColors.white;;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                    ...child.props,
                    style: { ...child.props.style, ...textStyle },
                    size: "footnote"
                })
                : child && child.type === Icon
                    ? React.cloneElement(child, {
                        ...child.props,
                        color: iconColor,
                        size: 15
                    })
                    : child && child.type === Image
                        ? React.cloneElement(child, {
                            ...child.props,
                            style: {
                                ...child.props.style,
                                width: 15,
                                height: 15
                            }
                        })
                        : null
        );

        return (
            <View
                ref={c => (this._root = c)}
                style={badgeStyle}
                {...otherProps}
            >
                {newChildren}
            </View>
        );
    }
}

Badge.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

Badge.defaultProps = {
    color: commonColors.primary,
};

const styles = StyleSheet.create({
    defaultBadge: {
        margin: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingHorizontal: 10,
        paddingVertical: 5
    },
});
