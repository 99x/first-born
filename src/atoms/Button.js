import React, { Component } from "react";
import { TouchableOpacity, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { getFontSize, getButtonPadding, getRoundRadius, getIconSize } from "../variables/buttonSizeVariables";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Button extends Component {
    render() {
        const { title, color, rounded, outline, block, size, secondary, transparent, ...otherProps } = this.props;

        const buttonColor = secondary ? commonColors.secondary : color;

        let buttonStyle = [styles.defaultButton];
        let iconSize = getIconSize(size);
        let textStyle;
        let iconColor;

        if (block) {
            buttonStyle.push(styles.blockButton);
        }

        if (rounded) {
            buttonStyle.push({ paddingHorizontal: getButtonPadding(size) + (getButtonPadding(size) / 3), borderRadius: getRoundRadius(size) });
        }

        if (outline) {
            textStyle = { color: buttonColor, fontSize: getFontSize(size) };
            buttonStyle.push({ borderColor: buttonColor, borderWidth: 1, padding: getButtonPadding(size) });
            iconColor = buttonColor;
        } else {
            textStyle = { color: commonColors.white, fontSize: getFontSize(size) };
            buttonStyle.push({ backgroundColor: buttonColor, padding: getButtonPadding(size) });
            iconColor = commonColors.white;
        }

        if (transparent) {
            buttonStyle.push(styles.transparent);
        }

        const children = React.Children.map(this.props.children, child => child && child.type === Text ?
            React.cloneElement(child, { style: textStyle, ...child.props }) : child && child.type === Icon ?
                React.cloneElement(child, { color: iconColor, size: iconSize, ...child.props }) : child && child.type === Image ?
                    React.cloneElement(child, { style: { width: 25, height: 25 }, ...child.props }) : null);

        return (
            <TouchableOpacity ref={(c) => this._root = c} style={buttonStyle} {...otherProps}>
                {children}
            </TouchableOpacity>
        )
    }
}

Button.propTypes = {
    rounded: PropTypes.bool,
    outline: PropTypes.bool,
    block: PropTypes.bool,
    title: PropTypes.string,
    color: PropTypes.string,
    transparent: PropTypes.bool,
    size: PropTypes.oneOf(["small", "default", "large"]),
    ...TouchableOpacity.propTypes
}

Button.defaultProps = {
    color: commonColors.primary,
    size: "default"
}

const styles = StyleSheet.create({
    text: {
        color: commonColors.white
    },
    defaultButton: {
        ...Platform.select({
            android: {
                borderRadius: 5,
                elevation: 3,
            },
            ios: {
                borderRadius: 10,
                shadowOffset: { width: 3, height: 3 },
                shadowOpacity: 0.3,
                shadowRadius: 3,
            }
        }),
        margin: 10,
        shadowColor: "#000",
    },
    blockButton: {
        width: "100%",
        justifyContent: "center",
        alignItems: "center",
    },
    transparent: {
        borderWidth: 0,
        backgroundColor: "transparent",
        borderRadius: 0,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 0,
        shadowRadius: 0,
        elevation: 0,
    }
})
