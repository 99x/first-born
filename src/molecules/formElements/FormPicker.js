import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../../atoms/Picker";
import { Text } from "../../atoms/Text";
import { commonColors } from "../../utils/color";

export class FormPicker extends Component {
    render() {
        const { label, children, data, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <Text size="sub_heading" color={commonColors.darkGrey}>
                    {label}
                </Text>
                <Picker {...otherProps}>
                    {children
                        ? children
                        : data
                        ? data.map((dataElement, key) => (
                              <Picker.Item {...dataElement} key={key} />
                          ))
                        : null}
                </Picker>
            </View>
        );
    }
}

FormPicker.propTypes = {
    label: PropTypes.string.isRequired,
    date: PropTypes.array,
    ...Picker.propTypes
};

const styles = StyleSheet.create({
    container: {
        width: "100%"
    }
});
