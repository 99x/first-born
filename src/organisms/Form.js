import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../atoms/Picker";

import { FormDatePicker } from "../molecules/formElements/FormDatePicker";
import { FormInput } from "../molecules/formElements/FormInput";
import { FormPicker } from "../molecules/formElements/FormPicker";
import { FormTextArea } from "../molecules/formElements/FormTextArea";

export class Form extends Component {
    render() {
        const { color, formElements, ...otherProps } = this.props;

        const inputColor = color ? color : "#486c86";

        return (
            <View style={styles.container}>
                {formElements.map((element, key) => {
                    return (
                        <View style={styles.formElementContainer} key={key}>
                            {this.renderElements(element, inputColor, otherProps)}
                        </View>
                    );
                })}
            </View>
        );
    }

    renderElements(element, color, styleProps) {
        element.color = color;
        const { type, pickerData, ...otherProps } = element;

        switch (type) {
            case "text":
                return <FormInput {...styleProps} {...otherProps} />;
            case "textarea":
                return <FormTextArea {...styleProps} {...otherProps} />;
            case "date":
                return <FormDatePicker {...styleProps} {...otherProps} />;
            case "picker":
                return (
                    <FormPicker {...styleProps} {...otherProps}>
                        {pickerData.map((dataElement, key) => (
                            <Picker.Item {...dataElement} key={key} />
                        ))}
                    </FormPicker>
                );
        }
    }
}

Form.propTypes = {
    color: PropTypes.string,
    activeStyle: PropTypes.object,
    errorStyle: PropTypes.object,
    rounded: PropTypes.bool,
    underline: PropTypes.bool,
    defaultStyle: PropTypes.bool,
    noStyle: PropTypes.bool,
    errorColor: PropTypes.string,
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: "100%"
    },
    formElementContainer: {
        marginTop: 10
    }
});
