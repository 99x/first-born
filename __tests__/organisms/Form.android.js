import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Form } from "../../src/organisms/Form";
import { FormPicker } from "../../src/molecules/formElements/FormPicker";
import { Picker } from "../../src/atoms/Picker";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

jest.mock("Picker", () => {
    // eslint-disable-next-line import/no-unresolved
    const React = require("React");
    const PropTypes = require("prop-types");
    return class MockPicker extends React.Component {
        static Item = props =>
            React.createElement("Item", props, props.children);
        static propTypes = { children: PropTypes.any };
        static defaultProps = { children: "" };

        render() {
            return React.createElement(
                "Picker",
                this.props,
                this.props.children
            );
        }
    };
});

it("renders a form with input types text", () => {
    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a form with input types, text and textarea", () => {
    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" },
        { label: "Address", type: "textarea" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a form with input types, text, textarea and picker", () => {
    const pickerData = [
        { value: "1", label: "1" },
        { value: "2", label: "2" },
        { value: "3", label: "3" }
    ];

    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" },
        { label: "Type", type: "picker", pickerData: pickerData },
        { label: "Address", type: "textarea" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a form with all input types", () => {
    const pickerData = [
        { value: "1", label: "1" },
        { value: "2", label: "2" },
        { value: "3", label: "3" }
    ];

    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" },
        { label: "Type", type: "picker", pickerData: pickerData },
        { label: "Date", type: "date" },
        { label: "Address", type: "textarea" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});
