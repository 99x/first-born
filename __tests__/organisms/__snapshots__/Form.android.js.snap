// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form with all input types 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          rejectResponderTermination={true}
          style={
            Object {
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Type
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <Picker
          mode="dropdown"
          onValueChange={[Function]}
          style={
            Object {
              "height": 45,
            }
          }
        >
          <Item
            label="1"
            value="1"
          />
          <Item
            label="2"
            value="2"
          />
          <Item
            label="3"
            value="3"
          />
        </Picker>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Date
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              },
              Object {
                "color": "rgba(33, 33, 33, 0.5)",
                "flex": 1,
              },
            ]
          }
        >
          Select Date
        </Text>
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "rgba(33, 33, 33, 0.5)",
                "fontSize": 20,
              },
              Object {
                "paddingRight": 15,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ïŠ«
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Address
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 100,
              "textAlignVertical": "top",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders a form with input types text 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          rejectResponderTermination={true}
          style={
            Object {
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders a form with input types, text and textarea 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          rejectResponderTermination={true}
          style={
            Object {
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Address
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 100,
              "textAlignVertical": "top",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders a form with input types, text, textarea and picker 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          rejectResponderTermination={true}
          style={
            Object {
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Type
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <Picker
          mode="dropdown"
          onValueChange={[Function]}
          style={
            Object {
              "height": 45,
            }
          }
        >
          <Item
            label="1"
            value="1"
          />
          <Item
            label="2"
            value="2"
          />
          <Item
            label="3"
            value="3"
          />
        </Picker>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(33, 33, 33, 0.87)",
              "fontSize": 15,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        Address
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 10,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "height": 100,
              "textAlignVertical": "top",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
